test/src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -321,6 +321,10 @@ public Jenkins getInstance() {
      * @throws Throwable if setup fails (which will disable {@code after}
      */
     public void before() throws Throwable {
+        if (Thread.interrupted()) { // JENKINS-30395
+            System.err.println("was interrupted before start");
```

Comment:
>`stderr` is unbuffered and this would be printed before anything else in the test.

Yes - but there are tests that have run before as reuseforks=true in surefire - and the logging in there may not have been completely flushed to the stream. JUL make no guarantees that a single log line will be written in a single call to system (even if the current implementation may do that) - or that log entry will not be written asynchronously.
SO what I am saying is there are no gaurantees that you will never get logs lines interspersed 

e.g.
`2016-01-01 05: 48 was interrupted before start INFO Tests ran OK`